pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -554,6 +554,12 @@ def _save_top_k_checkpoints(self, trainer, pl_module, metrics):
         epoch = metrics.get("epoch")
         step = metrics.get("step")
 
+        # when `val_loss` is being logged and no ModelCheckpoint is being provided
+        # `val_loss` will be selected for monitor and need to be reduced to
+        # prevent processes divergence
+        if self.monitor == "val_loss":
+            current = trainer.training_type_plugin.reduce(current, reduce_op="mean")
```

Comment:
Can you address this comment? From the other PR

https://github.com/PyTorchLightning/pytorch-lightning/pull/6004#discussion_r576778261

Even though this might fix the bug, it really doesn't seem like the place to  add this fix as it pollutes the ModelCheckpoint with unrelated logic and violates the call-hierarchy recently defined in the accelerator refactor.

cc: @awaelchli @justusschock

Reply:
I closed the other PR.

Reply:
Would be moved in another PR.